docs/content/concepts/io-management/unconnected-inputs.mdx

```diff
@@ -103,6 +101,143 @@ my_job.execute_in_process(
+### Using a input manager with subselection
+
+You might want to execute a subset of ops in your job and control how the inputs of those ops are loaded. Custom input managers also help in these situations, because the inputs at the beginning of the subset become unconnected inputs.
+
+For example, you might have `op1` that normally produces a table that `op2` consumes. To debug `op2`, you might want to run it on a different table than the one normally produced by `op1`.
+
+To accomplish this, you can set up the `input_manager_key` on `op2`'s `In` to point to an input manager with the desired loading behavior. As in the previous example, setting the `input_manager_key` on an `In` controls how that input is loaded and you can write custom loading logic. To ensure that the outputs of `op2` get stored as expected, you can write your new input manager as a subclass of the main IO manager.
+
+```python file=/concepts/io_management/input_managers.py startafter=start_load_input_subset endbefore=end_load_input_subset
+class MyIOManager(IOManager):
+    def handle_output(self, context, obj):
+        table_name = context.name
+        write_dataframe_to_table(name=table_name, dataframe=obj)
+
+    def load_input(self, context):
+        return read_dataframe_from_table(name=context.upstream_output.name)
+
+
+@io_manager
+def my_io_manager(_):
+    return MyIOManager()
+
+
+class MyInputLoader(MyIOManager):
+    def load_input(self, context):
+        return read_dataframe_from_table(name="table_1")
+
+
+@io_manager
+def my_input_loader():
+    return MyInputLoader()
+
+
+@op
+def op1():
+    """Do stuff"""
+
+
+@op(ins={"dataframe": In(input_manager_key="my_input_manager")})
+def op2(dataframe):
+    """Do stuff"""
+    dataframe.head()
+
+
+@job(
+    resource_defs={
+        "io_manager": my_io_manager,
+        "my_input_manager": my_input_loader,
+    }
+)
+def my_subselection_job():
+    op2(op1())
+```
+
+So far, this is set up so that `op2` always loads `table_1` even if you execute the full job. To write this so that `op2` only loads `table_1` when no input is provided from an upstream op, we can rewrite the input manager as follows:
+
+```python file=/concepts/io_management/input_managers.py startafter=start_better_load_input_subset endbefore=end_better_load_input_subset
+class MyNewInputLoader(MyIOManager):
+    def load_input(self, context):
+        if context.upstream_output is None:
+            return read_dataframe_from_table(name="table_1")
+        else:
+            return super().load_input(context)
+```
+
+Now, when running the full job, `op2`'s input will be loaded using the IO manager on the output of `op1`. When running the job subset, `op2`'s input has no upstream output, so `table_1` will be loaded.
+
+```python file=/concepts/io_management/input_managers.py startafter=start_execute_subselection endbefore=end_execute_subselection
+my_subselection_job.execute_in_process(
+    op_selection=["op2"],
+)
+```
+
+## Root Input Managers
```

Comment:
do we want to keep the docs for root input managers around as reference for people who already wrote them? or do we expect them to just pull up an older version of the docs?